games/main.tex

```diff
@@ -0,0 +1,1123 @@
+making them accountable in case of non-compliance (swindle).
+Such a system is playfully called a \emph{swap, swear and swindle} game.
+
+In this paper we present the generalised version of this idea and show how the paradigm
+can serve as the base layer infrastructure driving decentralised service economies.
+The tools presented here provide a platform to implement digital services that are
+cheap and scalable without compromising security offered by the public blockchain.
+The claim is that virtually any digital service can be reinterpreted as a swap, swear and swindle game.
+
+The solutions presented are built up in a piecemeal fashion from very simple and intuitive
+modules. Various errors in smart contracts revealed that turing completeness while expressive, can
+be dangerous and prompted many to call for solutions with less expressive power.
+Our modular approach as well as real world analogues of the components has the added
+advantage that reasoning about complex systems is easier
+and enable less error-prone implementations.
+
+In section 2 we introduce swap channels, an off-chain peer-to-peer accounting and
+payment solution for bidirectional services. We start from an off-chain protocol to
+account for service-for-service exchange between peers, introducing compensation for services
+and various ways to minimise blockchain interaction yet mitigate liability due to delayed payments.
+
+Section 3 introduces a taxonomy of promissory notes passed between peers in a swap channel.
+and shows how future payment promises can serve as enforcable service contracts.
+Conditional bonds pay out rewards upon successful delivery and
+implement positive incentivisation for decentralised services.
+
+In section 4, we present a way to implement service guarantees using an abstract
+challenge based system inspired by the blockchain as a judge paradigm. The threat
+of enforcable punitive measures serves as incentive to play by the rules.
+
+In section 5, we show how swap-channels can form a service network and extend the
+scope of economic interaction between peers both in terms of reach and frequency, yet
+preserve the security and scalability offered by swap-channels.
+
+Section 6 discusses price signalling and shows a way to eliminate the opportunity cost
+inherent in advance payments for future services when using appreciating assets like cryptocurrency.
+We conclude by showing how real world digital services can serve to back stablecoins.
+
+In the appendix you find the implementation details of smart contracts underpinning the
+system, as well as detailed examples of their application to data storage insurance and
+generic off-chain payments.
+
+\section{Swap channels}
+
+One of the major issues with direct onchain payments is that each transaction is processed
+by each node participating in the public network incurring high transaction costs.
+If a beneficiary is willing to trade off the risk of settlement and delay for decreased transaction costs
```

Comment:
Off-chain payments might expose all players (both beneficiaries and payers ) to a market risk due to the fluctuation nature of most of the tokens.
It might be nice to add few words about it in the doc